src/query/api/v1/handler/prometheus/remote/read.go

```diff
@@ -259,103 +260,119 @@ type ReadResult struct {
-			case *promql.VectorSelector:
-				if evalRange > 0 {
-					start = start.Add(-1 * evalRange)
-					evalRange = 0
-				}
-
-				if n.Offset > 0 {
-					offsetMilliseconds := time.Duration(n.Offset) * time.Millisecond
-					start = start.Add(-1 * offsetMilliseconds)
-					end = end.Add(-1 * offsetMilliseconds)
-				}
 
-				matchers, err := toLabelMatchers(n.LabelMatchers)
-				if err != nil {
-					return err
-				}
+		switch n := node.(type) {
+		case *promql.VectorSelector:
+			if evalRange > 0 {
+				start = start.Add(-1 * evalRange)
+				evalRange = 0
+			}
 
-				query := &prompb.Query{
-					StartTimestampMs: storage.TimeToPromTimestamp(start),
-					EndTimestampMs:   storage.TimeToPromTimestamp(end),
-					Matchers:         matchers,
-				}
+			if n.Offset > 0 {
+				offset := time.Duration(n.Offset)
+				start = start.Add(-1 * offset)
+				end = end.Add(-1 * offset)
+			}
 
-				req.Queries = append(req.Queries, query)
+			matchers, err := toLabelMatchers(n.LabelMatchers)
+			if err != nil {
+				return err
+			}
 
-			case *promql.MatrixSelector:
-				evalRange = n.Range
-				if evalRange > 0 {
-					start = start.Add(-1 * evalRange)
-					evalRange = 0
-				}
+			query := &prompb.Query{
+				StartTimestampMs: storage.TimeToPromTimestamp(start),
+				EndTimestampMs:   storage.TimeToPromTimestamp(end),
+				Matchers:         matchers,
+			}
 
-				if n.Offset > 0 {
-					offsetMilliseconds := time.Duration(n.Offset) * time.Millisecond
-					start = start.Add(-1 * offsetMilliseconds)
-					end = end.Add(-1 * offsetMilliseconds)
-				}
+			req.Queries = append(req.Queries, query)
 
-				matchers, err := toLabelMatchers(n.LabelMatchers)
-				if err != nil {
-					return err
-				}
+		case *promql.MatrixSelector:
+			evalRange = n.Range
+			if evalRange > 0 {
+				start = start.Add(-1 * evalRange)
+				evalRange = 0
+			}
 
-				query := &prompb.Query{
-					StartTimestampMs: storage.TimeToPromTimestamp(start),
-					EndTimestampMs:   storage.TimeToPromTimestamp(end),
-					Matchers:         matchers,
-				}
+			if n.Offset > 0 {
+				offset := time.Duration(n.Offset)
+				start = start.Add(-1 * offset)
+				end = end.Add(-1 * offset)
+			}
 
-				req.Queries = append(req.Queries, query)
+			matchers, err := toLabelMatchers(n.LabelMatchers)
+			if err != nil {
+				return err
+			}
 
+			query := &prompb.Query{
+				StartTimestampMs: storage.TimeToPromTimestamp(start),
+				EndTimestampMs:   storage.TimeToPromTimestamp(end),
+				Matchers:         matchers,
 			}
-			return nil
-		})
+
+			req.Queries = append(req.Queries, query)
+		}
+
+		return nil
+	})
+
+	return req, nil
+}
+
+// ParseRequest parses the compressed request
+func ParseRequest(
+	ctx context.Context,
+	r *http.Request,
+	opts options.HandlerOptions,
+) (*prompb.ReadRequest, *storage.FetchOptions, *xhttp.ParseError) {
+	var req *prompb.ReadRequest
+	var rErr *xhttp.ParseError
+	switch {
+	case r.Method == http.MethodGet && strings.TrimSpace(r.FormValue("expr")) != "":
```

Comment:
Do we have to change this to "query" or what have you to make it compatible with the PromQL query client library methods?